### PR TITLE
refactor: refactor Connector to make code more expressive

### PR DESCRIPTION
**Pull request by wujunzhuo**
This change is [Reviewable](https://reviewable.io/reviews/yomorun/yomo/329)

### CODE REVIEW

File: router.go [hunk 1]
```diff
@@ -1,93 +1,90 @@
...
+func (r *route) GetForwardRoutes(tag byte) []string {
+	r.mu.RLock()
+	defer r.mu.RUnlock()
 
-	routes := make([]string, 0)
-	r.data.Range(func(key interface{}, val interface{}) bool {
-		if key.(int) > idx {
-			routes = append(routes, val.(string))
+	if conns := r.data[tag]; conns != nil {
+		keys := make([]string, 0, len(conns))
```
Comment (fanweixiao) [inline router.go:83]: consider `var keys []string ...... return keys`

File: app_info.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
+package yomo
+
+import (
+	"github.com/yomorun/yomo/core"
+	"github.com/yomorun/yomo/core/frame"
+)
+
+type appInfo struct{}
```
Comment (venjiang) [inline app_info.go:8]: `appInfo` name makes me feel weird.

Comment (wujunzhuo) [reply app_info.go:8]: probably `ConnInfo` might be better?

File: router.go [hunk 3]
```diff
@@ -1,93 +1,90 @@
...
 func (r *router) Clean() {
-	r.config = nil
+	r.r = nil
 }
 
-// route interface
 type route struct {
-	data sync.Map
+	functions []config.App
+	data      map[byte]map[string]struct{}
```
[the remark this answers is not in the document]
Comment (wujunzhuo) [reply router.go:29]: yes, but it looks hard to maintain backward compatibility

**Review by fanweixiao — APPROVED** [review]

File: router.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (fanweixiao) [reply router.go:29]: You are right. So let's do it from v2

File: core/server.go [hunk 4]
```diff
@@ -283,76 +278,57 @@ func (s *Server) handleHandshakeFrame(c *Context) error {
...
+	if err != nil {
+		return err
+	}
+
 	// route
 	if err := s.validateRouter(); err != nil {
 		return err
 	}
-	route := s.router.Route()
-	if reflect.ValueOf(route).IsNil() {
```
Comment (venjiang) [inline core/server.go:300]: suggest reserved

File: app_info.go (hunk 2 above)
Comment (fanweixiao) [reply app_info.go:8]: `AppInfo` -> `ConnState`

File: meta_data.go [hunk 5]
```diff
@@ -0,0 +1,30 @@
+package yomo
+
+import (
+	"github.com/yomorun/yomo/core"
+	"github.com/yomorun/yomo/core/frame"
+)
+
+type metaData struct{}
```
Comment (venjiang) [inline meta_data.go:8]: metaData -> metadata, It's a word.

File: core/frame/frame.go [hunk 6]
```diff
@@ -16,7 +16,7 @@ const (
 	TagOfMetaFrame     Type = 0x2F
 	TagOfMetadata      Type = 0x03
 	TagOfTransactionID Type = 0x01
-	TagOfIssuer        Type = 0x02
+	TagOfMetaData      Type = 0x02
```
Comment (venjiang) [inline core/frame/frame.go:19]: `TagOfMetadata` is no longer used and can be used directly

File: core/frame/meta_frame.go [hunk 7]
```diff
@@ -30,14 +34,30 @@ func (m *MetaFrame) TransactionID() string {
 	return m.tid
 }
 
+// SetMetaData set the extra info of the application
+func (m *MetaFrame) SetMetaData(data []byte) {
```
Comment (venjiang) [inline core/frame/meta_frame.go:38]: `SetMetadata` is recommended.

File: core/frame/meta_frame.go [hunk 8]
```diff
@@ -30,14 +34,30 @@ func (m *MetaFrame) TransactionID() string {
...
+	m.data = data
+}
+
+// MetaData returns the extra info of the application
+func (m *MetaFrame) MetaData() []byte {
```
Comment (venjiang) [inline core/frame/meta_frame.go:43]: `Metadata()` is recommended.

File: app_info.go (hunk 2 above)
Comment (wujunzhuo) [reply app_info.go]: renamed to `Metadata`

File: meta_data.go (hunk 5 above)
Comment (wujunzhuo) [reply meta_data.go]: done

**Review by venjiang — APPROVED** [review]

**Pull request merged**